compiler-rt/lib/asan/asan_poisoning.h

```diff
@@ -19,9 +19,25 @@
 #include "asan_mapping.h"
 #include "sanitizer_common/sanitizer_flags.h"
 #include "sanitizer_common/sanitizer_platform.h"
+#include "sanitizer_common/sanitizer_ring_buffer.h"
 
 namespace __asan {
 
+struct PoisonRecord {
+  unsigned int stack_id;
+  unsigned int thread_id;
+  uptr begin;
+  uptr end;
+};
+
+using PoisonRecordRingBuffer = RingBuffer<struct PoisonRecord>;
+
+// Set up data structures for track_poison.
+void InitializePoisonTracking();
+
+PoisonRecordRingBuffer* AcquirePoisonRecords();
```

Comment:
I mean proper API :)

void AddPoisonRecord(const PoisonRecord&);
const PoisonRecord& FindPoisonRecord(uptr begin, uptr end);

Reply:
I added AddPoisonRecord and FindPoisonRecord. For FindPoisonRecord, since this may fail to find a match (e.g., it aged out of the ring buffer), I used `bool FindPoisonRecord(uptr addr, const PoisonRecord& match)`.

I kept the Acquire/Release API for the fork handler.